src/pybind/mgr/cephadm/module.py

```diff
@@ -1456,35 +1456,68 @@ def add_host(self, spec: HostSpec) -> str:
         return self._add_host(spec)
 
     @handle_orch_error
-    def remove_host(self, host):
-        # type: (str) -> str
+    def remove_host(self, host: str, force: bool = False, offline: bool = False) -> str:
         """
         Remove a host from orchestrator management.
 
         :param host: host name
+        :param force: bypass runnning daemons check
+        :param offline: remove offline host
         """
-        # Verify if it is possible to remove the host safely
-        daemons = self.cache.get_daemons_by_host(host)
-        if daemons:
-            self.log.warning(f"Blocked {host} removal. Daemons running: {daemons}")
 
-            daemons_table = ""
-            daemons_table += "{:<20} {:<15}\n".format("type", "id")
-            daemons_table += "{:<20} {:<15}\n".format("-" * 20, "-" * 15)
+        # check if offline
+        if not offline:
+            offline = host in self.offline_hosts or self.inventory._inventory[host].get(
+                'status', '').lower() == 'offline'
```

Comment:
IMO, setting this ourselves here defeats the purpose of having the flag. If a user doesn't provide the offline flag when removing an offline host we'll just set offline to be True so why ever bother setting the flag? I feel like with this flag we should just check if it has been set correctly (i.e. it is set if the host to remove is offline, not set if host to remove is not offline) and just bail out and return a warning message if it was set wrong. That way we force the user to explicitly recognize that they are removing an offline host. If we don't care if they know if the host they're removing is offline we could just remove the flag from the cli and set it ourselves every time.
```suggestion
        is_offline = host in self.offline_hosts or self.inventory._inventory[host].get(
                'status', '').lower() == 'offline'
        if is_offline != offline:
            Return Error Message
```